functions/src/index.ts

```diff
@@ -257,6 +382,57 @@ const generateReadingsCsvRuntimeOptions: functions.RuntimeOptions = {
   timeoutSeconds: 540,
 };
 
+/**
+ * This function populates the given sensor doc with a default circular buffer
+ * for eitherAQI or PM2.5
+ * @param aqiBuffer - true if AQI buffer, false if PM25 buffer
+ * @param docRef - document reference for the sensor to update
+ */
+async function populateBuffer(
+  aqiBuffer: boolean,
+  docRef: FirebaseFirestore.DocumentReference<FirebaseFirestore.DocumentData>
+) {
+  let bufferSize = 0;
+  const bufferIndex = 0;
+  if (aqiBuffer) {
+    // 144 = (6 calls/hour * 24 hours) is the amount of entries we need to
+    // create a graph with 24 hours of data
+    bufferSize = 144; /* eslint-disable-line no-magic-numbers */
+    const aqiBuffer: Array<AQIBufferElement> = [];
```

Comment:
If we know the size, I feel like we can be a bit more efficient by declaring size up front and using array methods to fill.
```suggestion
    const aqiBuffer: Array<AQIBufferElement> = Array(bufferSize).fill(defaultAQIBufferElement);
 ```
    
    
```